tools/gradle-plugin/src/test/kotlin/io/github/composegears/valkyrie/gradle/ValkyrieGradlePluginTest.kt

```diff
@@ -0,0 +1,349 @@
+        // given
+        root.resolve("build.gradle.kts").writeText(
+            """
+                plugins {
+                    kotlin("jvm")
+                    id("io.github.composegears.valkyrie")
+                }
+
+                valkyrie {
+                    packageName = "x.y.z"
+                }
+            """.trimIndent(),
+        )
+        root.writeTestSvgs(sourceSet = "main")
+
+        // when
+        val result = runTask(root, TASK_NAME).build()
+
+        // then
+        assertThat(result.task(":$TASK_NAME")?.outcome).isEqualTo(SUCCESS)
+        assertThat(result.output).contains("Generated 4 ImageVectors in package x.y.z")
+        assertThat(result.output).contains("LinearGradient.kt")
+        assertThat(result.output).contains("RadialGradient.kt")
+        assertThat(result.output).contains("ClipPathGradient.kt")
+        assertThat(result.output).contains("LinearGradientWithStroke.kt")
+    }
+
+    @Test
+    fun `Generate from test SVGs with custom config`() {
+        // given
+        root.resolve("build.gradle.kts").writeText(
+            """
+                import io.github.composegears.valkyrie.generator.jvm.imagevector.OutputFormat
+                import io.github.composegears.valkyrie.generator.jvm.imagevector.PreviewAnnotationType
+
+                plugins {
+                    kotlin("jvm")
+                    id("io.github.composegears.valkyrie")
+                }
+
+                valkyrie {
+                    packageName = "x.y.z"
+                    iconPackName = "MyIconPack"
+                    nestedPackName = "MyNestedPack"
+                    outputFormat = OutputFormat.LazyProperty
+                    useComposeColors = false
+                    generatePreview = true
+                    previewAnnotationType = PreviewAnnotationType.Jetbrains
+                    useFlatPackage = true
+                    useExplicitMode = true
+                    addTrailingComma = true
+                    indentSize = 8
+                }
+            """.trimIndent(),
+        )
+        root.writeTestSvgs(sourceSet = "main")
+
+        // when
+        val result = runTask(root, TASK_NAME).build()
+
+        // then the expected files are printed to log
+        assertThat(result.task(":$TASK_NAME")?.outcome).isEqualTo(SUCCESS)
+        assertThat(result.output).contains("Generated 4 ImageVectors in package x.y.z")
+        assertThat(result.output).contains("LinearGradient.kt")
+        assertThat(result.output).contains("RadialGradient.kt")
+        assertThat(result.output).contains("ClipPathGradient.kt")
+        assertThat(result.output).contains("LinearGradientWithStroke.kt")
+
+        // and the LinearGradient file is created with the right visibility, parent pack, nested pack, etc
+        val linearGradientKt = project
+            .fileTree(root)
+            .first { it.name == "LinearGradient.kt" }
+            .readText()
```

Comment:
- `project` is not shared, we don't need it to be a property.
- In this case, you want `DefaultProject`, don't call the internal API, try something like `walk` or `listDirectoryEntries` for the specific dir.